tests/favorites.spec.js

```diff
@@ -227,5 +227,58 @@ describe('Test the favorites route', () => {
       expect(res.body).toHaveProperty("rating", null);
 
     })
+
+    test('It will not add duplicate favorites', async() => {
+      let newFav = {
+        title: "We will Rock You",
+        artistName: 'Queen'
+      }
+      // set a mock object and stub the fetch call to return a custom object
+      // so your fetch call always returns exactly what you want it to return
+      await fetch.mockResponse(
+        JSON.stringify({
+          message: {
+            body: {
+              track_list: [
+                {
+                  track: {
+                    artist_name: "Queen",
+                    track_name: "We Will Rock You",
+                    track_rating: 87,
+                    primary_genres: {
+                      music_genre_list: [{
+                        music_genre: {
+                          music_genre_name: "Rock"
+                        }
+                      }]
+                    }
+                  }
+                }
+              ]
+            }
+          }
+        })
+      );
+
+      const res_1 = await request(app)
+        .post("/api/v1/favorites")
+        .send(newFav);
+
+      const res_2 = await request(app)
+        .post("/api/v1/favorites")
+        .send(newFav);
+
+      let db_favorites = await database('favorites').select()
+        .then(result => {
+          return result;
+        })
```

Comment:
This definitely works. A more explicit test might try to find all instances in the db with the potentially duplicate information and make sure that length is 1. Since we're resetting the test db in between, this works for me too.

Reply:
Yeah and I think that if we had multiple different records that would be the way to go. As is this should work unless something changes in our tests.